File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        num_stack = []
        op_stack = []
        res = 0
        for i in tokens:
            if i == "+":
                num_stack.append(num_stack.pop() + num_stack.pop())
                continue
            if i == "/":
                a = num_stack.pop()
                b = num_stack.pop()
                num_stack.append(int(b / a))
                continue
            if i == "*":
                num_stack.append(num_stack.pop() * num_stack.pop())
                continue
            if i == "-":
                a = num_stack.pop()
                b = num_stack.pop()
                num_stack.append(b - a)
                continue
            else:
                num_stack.append(int(i))
        return int(num_stack[0])      


        

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
import java.util.Stack;

class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> numStack = new Stack<>();
        int res = 0;
        for (String token : tokens) {
            if (token.equals("+")) {
                numStack.push(numStack.pop() + numStack.pop());
                continue;
            }
            if (token.equals("/")) {
                int a = numStack.pop();
                int b = numStack.pop();
                numStack.push(b / a);
                continue;
            }
            if (token.equals("*")) {
                numStack.push(numStack.pop() * numStack.pop());
                continue;
            }
            if (token.equals("-")) {
                int a = numStack.pop();
                int b = numStack.pop();
                numStack.push(b - a);
                continue;
            } else {
                numStack.push(Integer.parseInt(token));
            }
        }
        return numStack.pop();
    }
}
